src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/FontCache/FontSource.cs

```diff
@@ -194,27 +194,27 @@ public UnmanagedMemoryStream GetUnmanagedStream()
                 {
                     WebResponse response = WpfWebRequestHelper.CreateRequestAndGetResponse(_fontUri);
                     fontStream = response.GetResponseStream();
-                    if (String.Equals(response.ContentType, ObfuscatedContentType, StringComparison.Ordinal))
+                    if (string.Equals(response.ContentType, ObfuscatedContentType, StringComparison.Ordinal))
                     {
                         // The third parameter makes sure the original stream is closed
                         // when the deobfuscating stream is disposed.
                         fontStream = new DeobfuscatingStream(fontStream, _fontUri, false);
                     }
                 }
 
-                UnmanagedMemoryStream unmanagedStream = fontStream as UnmanagedMemoryStream;
-
-                if (unmanagedStream != null)
-                    return unmanagedStream;
+                // We don't want any memory leaks
+                // TODO: Remove FinalizableUnmanagedStream once FontFileStream is migrated from C++/CLI.
+                if (fontStream is UnmanagedMemoryStream unmanagedMemoryStream)
+                    return new FinalizableUnmanagedStream(unmanagedMemoryStream);
 
+                // Convert the DeobfuscatingStream to byte[]; add it to our cache, dispose it
                 bits = StreamToByteArray(fontStream);
+                lock (_resourceCache)
+                {
+                    _resourceCache.Add(_fontUri, bits, false);
+                }
 
-                fontStream?.Close();
-            }
-
-            lock (_resourceCache)
-            {
-                _resourceCache.Add(_fontUri, bits, false);
+                fontStream.Close();
```

Comment:
GetCompositeFontResourceStream() could return null from ResourceManager.GetStream() if I see correctly. Although when that happens, StreamToByteArray crashes already, so I guess removing the null check here is fine.

Reply:
Yeah, the null-check on that and in `GetCompositeFontResourceStream` on freshly created RM were weird to say at the least, though you shouldn't be able to init a `FontSource` that's a composite font without it actually existing iirc.

Fixing a possible NRE shall be handled in a different PR if we want to be on the safe side imho.